File: server/lib/services-catalog.js
'use strict';

/**
 * Service codes team members can offer (aligned with deploy pricing / booking form).
 */
const OFFERABLE_SERVICES_CATALOG = [
    {
        section: 'Orthopaedic procedures',
        items: [
            { id: 'fracture_repair_simple', label: 'Fracture Repair (Simple)' },
            { id: 'fracture_repair_complex', label: 'Fracture Repair (Complex)' },
            { id: 'tplo', label: 'Tibial Plateau Levelling Osteotomy (TPLO)' },
            { id: 'fho', label: 'Femoral Head and Neck Excision (FHO)' },
            { id: 'medial_patella_luxation_repair', label: 'Medial Patella Luxation Repair' },
            { id: 'carpal_arthrodesis', label: 'Carpal Arthrodesis' },
            { id: 'tarsal_arthrodesis', label: 'Tarsal Arthrodesis' },
            { id: 'hif_repair', label: 'Humeral Intracondylar Fissure (HIF) Repair' },
            { id: 'angular_limb_deformity_correction', label: 'Angular Limb Deformity Correction' }
        ]
    },
    {
        section: 'Soft tissue procedures',
        items: [
            { id: 'perineal_urethrostomy', label: 'Perineal Urethrostomy' },
            { id: 'tecabo', label: 'Total Ear Canal Ablation & Bulla Osteotomy (TECABO)' },
            { id: 'mass_excision_simple', label: 'Mass Excision (Simple)' },
            { id: 'mass_excision_complex', label: 'Mass Excision with Complex Reconstruction' },
            { id: 'diaphragmatic_hernia_repair', label: 'Diaphragmatic Hernia Repair' },
            { id: 'perineal_hernia_repair', label: 'Perineal Hernia Repair' },
            { id: 'nephrectomy', label: 'Nephrectomy' },
            { id: 'liver_lobectomy', label: 'Liver Lobectomy' }
        ]
    },
    {
        section: 'Minimally invasive',
        items: [{ id: 'laparoscopic_surgery', label: 'Laparoscopic surgery' }]
    },
    {
        section: 'Diagnostic services',
        items: [
            { id: 'ultrasonography_abdominal', label: 'Ultrasonography (Abdominal)' },
            { id: 'echocardiography', label: 'Echocardiography' },
            { id: 'endoscopy_gastroscopy_colonoscopy', label: 'Endoscopy (Gastroscopy/Colonoscopy)' },
            { id: 'radiographic_interpretation', label: 'Radiographic Interpretation (Member Practices)' },
            { id: 'consultation_advice', label: 'Consultation & Advice (Member Practices)' }
        ]
    },
    {
        section: 'Other',
        items: [{ id: 'veterinary_locum', label: 'Veterinary locum' }]
    }
];

const OFFERABLE_SERVICE_IDS = OFFERABLE_SERVICES_CATALOG.flatMap(s => s.items.map(i => i.id));

module.exports = {
    OFFERABLE_SERVICES_CATALOG,
    OFFERABLE_SERVICE_IDS
};

File: src/main/java/uk/co/northernveterinaryservice/util/ServicesCatalog.java
package uk.co.northernveterinaryservice.util;

import java.util.List;
import java.util.Set;

/**
 * Service codes team members can offer — mirrors server/lib/services-catalog.js.
 */
public final class ServicesCatalog {

    private ServicesCatalog() {}

    public static final List<String> OFFERABLE_SERVICE_IDS = List.of(
        "fracture_repair_simple",
        "fracture_repair_complex",
        "tplo",
        "fho",
        "medial_patella_luxation_repair",
        "carpal_arthrodesis",
        "tarsal_arthrodesis",
        "hif_repair",
        "angular_limb_deformity_correction",
        "perineal_urethrostomy",
        "tecabo",
        "mass_excision_simple",
        "mass_excision_complex",
        "diaphragmatic_hernia_repair",
        "perineal_hernia_repair",
        "nephrectomy",
        "liver_lobectomy",
        "laparoscopic_surgery",
        "ultrasonography_abdominal",
        "echocardiography",
        "endoscopy_gastroscopy_colonoscopy",
        "radiographic_interpretation",
        "consultation_advice",
        "veterinary_locum"
    );

    public static final Set<String> OFFERABLE_SERVICE_IDS_SET = Set.copyOf(OFFERABLE_SERVICE_IDS);
}
